scripts/get-plugin-types.ts

```diff
@@ -0,0 +1,40 @@
+import { flatten, globP, mkdirP, ncpP, readFileP, spawnP, tmpDirP, writeFileP } from './utils';
+
+const TYPES_DIRECTORY = 'src/plugins/types';
+
+async function mkdirSafeP(dir: string) {
+  try {
+    await mkdirP(dir);
+    return true;
+  } catch (e) {
+    if (e.code !== 'EEXIST') {
+      throw new Error(`Error creating directory ${dir}`);
+    }
+    return false;
+  }
+}
+
+export async function getPluginTypes() {
+  await mkdirSafeP(TYPES_DIRECTORY);
+
+  const indexTs = (await readFileP(`${TYPES_DIRECTORY}/index.d.ts`, 'utf8') as string)
+    .split('\n');
+  for (const line of indexTs) {
+    const matches = line.match(/^import \* as .* from '\.\/(.*)';\s*\/\/\s*(.*)@(.*)$/);
```

Comment:
I think the regex's for `packageName`, `name`, and `version` must use `(.+)` rather than `(.*)`